internal/driver/custommetadata_test.go

```diff
@@ -34,244 +28,236 @@ func TestOnvifClient_getCustomMetadata(t *testing.T) {
 		},
 	}
 
-	singleDataTest := testCase{
-		data: `{
-		"CustomMetadata":[
-			"CommonName"
-		]
-		}`,
-		expectedResponse: contract.ProtocolProperties{
-			"CommonName": myDevice.Protocols[CustomMetadata]["CommonName"],
-		},
-	}
-
-	happyMultipleDataTest := testCase{
-		data: `{
-		"CustomMetadata":[
-			"Location",
-			"CommonName"
-		]
-		}`,
-		expectedResponse: contract.ProtocolProperties{
-			"Location":   myDevice.Protocols[CustomMetadata]["Location"],
-			"CommonName": myDevice.Protocols[CustomMetadata]["CommonName"],
-		},
-	}
-
-	noFieldDataTest := testCase{
-		data: `{
-		"CustomMetadata":["Movie"]
-		}`,
-		expectedResponse: contract.ProtocolProperties{
-			"Movie": "This field does not exist in custom metadata",
-		},
-	}
-
-	noFieldsDataTest := testCase{
-		data: `{
-		"CustomMetadata":["Movie", "Height"]
-		}`,
-		expectedResponse: contract.ProtocolProperties{
-			"Movie":  "This field does not exist in custom metadata",
-			"Height": "This field does not exist in custom metadata",
-		},
-	}
-	emptyData := testCase{
-		data:             `{"CustomMetadata":[]}`,
-		expectedResponse: contract.ProtocolProperties{},
-	}
-
 	tests := []struct {
 		name          string
 		device        contract.Device
 		data          string
 		expected      contract.ProtocolProperties
 		errorExpected bool
 	}{
-		{"happy path without data - getAll", myDevice, "", myDevice.Protocols[CustomMetadata], false},
-		{"happy path with data for single field", myDevice, singleDataTest.data, singleDataTest.expectedResponse, false},
-		{"happy path with data for multiple field", myDevice, happyMultipleDataTest.data, happyMultipleDataTest.expectedResponse, false},
-		{"happy path with data for non-existent field", myDevice, noFieldDataTest.data, noFieldDataTest.expectedResponse, false},
-		{"happy path with data for multiple non-existent fields", myDevice, noFieldsDataTest.data, noFieldsDataTest.expectedResponse, false},
-		{"with empty data", myDevice, emptyData.data, emptyData.expectedResponse, true},
-		{"badJson", myDevice, "bogus", myDevice.Protocols[CustomMetadata], true},
+		{
+			"happy path without data - getAll",
+			myDevice,
+			"",
+			myDevice.Protocols[CustomMetadata],
+			false},
+		{
+			"happy path with data for single field",
+			myDevice,
+			`{"CustomMetadata":["CommonName"]}`,
+			contract.ProtocolProperties{
+				"CommonName": myDevice.Protocols[CustomMetadata]["CommonName"],
+			},
+			false,
+		},
+		{
+			"happy path with data for multiple field",
+			myDevice,
+			`{"CustomMetadata":["Location","CommonName"]}`,
+			contract.ProtocolProperties{
+				"Location":   myDevice.Protocols[CustomMetadata]["Location"],
+				"CommonName": myDevice.Protocols[CustomMetadata]["CommonName"],
+			},
+			false,
```

Comment:
Instead of setting it to false using `errorExpected: false` for the happy paths, i would just leave that field out altogether (it is false by default anyways). 

This reduces the amount of code you have to write and makes it more noticeable when it is true, because your eyes will notice the `errorExpected: true` that wasn't there in the others.